tools/importTool/src/main/scala/AkkaFileReaderWithActor.scala

```diff
@@ -0,0 +1,161 @@
+      {
+        case Success(r) => println("Read the whole file from server successfully")
+        case Failure(e) => {
+          println("oopss...Got a failure in get request")
+          system.scheduler.scheduleOnce(7000 milliseconds, self, ActorInput(inputUrl, format))
+        }})
+    val graphResult = httpResponse.map ({ response =>
+      println("Get response status=" + response.status)
+      val hashedBody = response.headers.find(_.name == "Content-Range").map(_.value).getOrElse("")
+      response.entity.withSizeLimit(hashedBody.split("/")(1).toLong + 1).dataBytes
+        .via(Framing.delimiter(ByteString(System.lineSeparator()), 100000, true))
+        .map(_.utf8String)
+        .map(x => (getTripleKey(x), x))
+        .sliding(2, 1)
+        .splitAfter(pair => (pair.headOption, pair.lastOption) match {
+        case (Some((key1, _)), Some((key2, _))) => key1 != key2
+      })
+        .mapConcat(_.headOption.toList)
+        .fold("" -> List.empty[String]) {
+        case ((_, values), (key, value)) => key -> (value :: values)
+      }
+        .map(x => x._2)
+        .mergeSubstreams
+        .grouped(25)
+        .mapAsync(1)(postWithRetry)
+        .statefulMapConcat {
+          () => {
+            x => {
+              bytesAccumulatorActor ! Message(x._2)
+              List.empty
+            }
+          }
+        }
+        .runWith(Sink.head)
+        .onComplete(
+          {
+            case Success(r) => {
+              println("Process completed successfully")
+              system.terminate()
+            }
+            case Failure(e) => {
+              println("oopss...Got a failure in get request")
+              system.scheduler.scheduleOnce(7000 milliseconds, self, ActorInput(inputUrl, format))
+            }})
+
+    })
+    graphResult
+  }
+
+  def postWithRetry(batch: Seq[List[String]]): Future[(HttpResponse, Int)] = {
+    retry(10.seconds, 3){ ingest(batch) }
+  }
+
+
+  def retry[T](/*f: Seq[String] => Future[T], batch: => Seq[String],*/ delay: FiniteDuration, retries: Int)(task: => Future[(T, Int)])(implicit ec: ExecutionContext, scheduler: Scheduler): Future[(T, Int)] = {
+    task.recoverWith {
+      case e: Throwable if retries > 0 =>
+        println("Going to retry, retry count=" + retries)
+        println("Failed to retry post request,", e.getMessage)
+        akka.pattern.after(delay, scheduler)(retry(delay, retries - 1)(task))
+    }
+
+  }
+
+  def getTripleKey(triple:String): String = {
+    triple.split(" ")(0)
+  }
+
+  def ingest(batch: Seq[List[String]]): Future[(HttpResponse, Int)] = {
+    var infotonsList = batch.flatten
+    println("Infotons " + infotonsList)
+    println("great,infotons batch size= " + batch.size + ", batch count lines=" + infotonsList.size)
+    val batchSizeInBytes = infotonsList.mkString.getBytes.length + infotonsList.size
+    var postRequest = HttpRequest(
+      HttpMethods.POST,
+      "http://localhost:9000/_in?format=" + format,
```

Comment:
This is a design bug. And my fault as well :) Please expose the **target cluster** (you're ingesting to) as a scallop parameter: `--cluster` or `--target`

The reason is we want to allow this external tool to be running from a machine which is not necessarily a CM-Well node.

Reply:
done